### ISSUE TITLE
aria-multiselectable is not valid for table elements

### ISSUE THREAD

**Issue by derick-montague** [bootstrap-vue/bootstrap-vue#6305]
### Describe the bug

When creating a table that has select-able rows, the aria-multiselectable attribute is added to the table element. This attribute is not valid on table elements.

### Steps to reproduce the bug

1. Create a bootstrap-table
2. Add the selectable prop
3. Test accessibility with Deque Axe, IBM Accessibility Checker, or Lighthouse

### Expected behavior

There shouldn't be a violation.

### Versions

**Libraries:**

- BootstrapVue: 2.21.2
- Bootstrap: 4.5.0
- Vue: 2.6.11

**Environment:**

- Device: All browsers
- OS: Any
- Browser: All
- Version: Any

### Demo link

https://codepen.io/BeyondHyper/pen/xxEJXXG

### Additional context

Open the pen in debug mode and run Deque Axe, IBM Accessibility Checker, or Lighthouse w/Accessibility checked to review the violation.

**Comment by andrei-gheorghiu** [comment]
> This attribute is not valid on table elements.

Could you back up your claim with a link to any normative documentation? (Hint: you can't, because the normative documentation states the opposite).

**Comment by derick-montague** [comment]
Thank you for editing the initial post with the "hint that no documentation exists" comment. This is a great library and I appreciate all the hard work from its contributors. 

The error was found running all 3 of the accessibility browser plugins, Axe, Lighthouse, and IBM's Accessibility checker.

I am not sure the value it provides to a screen reader when applied to an HTML table element. The uses documented in the WAI-ARIA guidelines on W3 (https://www.w3.org/TR/wai-aria-1.0/states_and_properties#aria-multiselectable) state the use to be associated with the roles grid, listbox, tablist, and tree roles.

If the ARIA role is not useful, its better not to use it. No ARIA is better than bad ARIA.

**Comment by jackmu95** [comment]
I've created a CodeSandbox with the same code of the CodePen since the debug mode expired:
https://codesandbox.io/s/strange-cache-p3pim

**Comment by andrei-gheorghiu** [comment]
@derick-montagne, you probably misunderstood my previous message. 
At the same time, you seem to have misunderstood the normative which you linked.

Before I made my comment, I looked at the same exact normative you linked and I couldn't find where it says that 

    <table role="grid" aria-multiselectable="true">...</table>
    
... is invalid.

More so, in the MDN page dedicated to [ARIA: grid role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Grid_Role), they're featuring a `<table>` as example.

**Comment by derick-montague** [comment]
> @derick-montagne, I asked for more details because your initial statement
> 
> > This attribute is not valid on table elements.
> 
> was incomplete, if not untrue. In order to make it true, you have to add
> 
> > _... without a `role` attribute set to `grid`, `listbox`, `tablist` or `tree`_
> 
> to it.
> 
> I was intrigued because the MDN page dedicated to [ARIA: grid role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Grid_Role), lists a `<table>` as example.

I agree that it becomes a challenge to make these types of determinations. I am not an accessibility expert. I assume since a table is not listed in the W3 documentation, that means that the use of `aria-multiselectable` to be valid only with an element assigned one of those roles. The reason I was researching was due to the 3 common accessibility checkers, Axe by Deque, Accessibility Checker by IBM, and Lighthouse all reported this code as a violation. All three report the same violation. If only one of them reported it, I would have assumed it may be a bug in the tool. However in this case, I expect the violation to be accurate. These are reports from the 3 tools:

## Accessibility Checker
> The WAI-ARIA role or attribute 'aria-multiselectable' is not valid for the element `<table>`

## Axe
> Elements must only use allowed ARIA attributes
> https://dequeuniversity.com/rules/axe/4.1/aria-allowed-attr?application=AxeChrome

## Lighthouse
> [aria-*] attributes do not match their roles
> Each ARIA `role` supports a specific subset of `aria-*` attributes. Mismatching these invalidates the `aria-*` attributes. 
> https://web.dev/aria-allowed-attr/?utm_source=lighthouse&utm_medium=devtools

Regarding the use of `role="grid"` to fix the issue, it does remove the violation in all the tools. I do wonder if it is the right way to resolve the issue. [Adrian Roselli suggests grid is an anti-pattern](
https://adrianroselli.com/2020/07/aria-grid-as-an-anti-pattern.html). [W3 says when using aria-multiselectable, we also need to use the `aria-selected` attribute on all selectable descendants](https://www.w3.org/TR/wai-aria-1.0/states_and_properties#aria-multiselectable). That is not something that's being added currently. I don't understand how valuable the `aria-multiselectable` attribute is since I am not an expert and don't have the ability to test with users of varying abilities.

**Comment by andrei-gheorghiu** [comment]
Nobody is an expert in accessibility, because it's a work in progress. Not even the guys making the standard. But they're doing their best. 
Since you opened the issue I looked closer into it. 
I'd say the major problem is in how those plugins report the bug. They simply state: this attribute is illegal on this element. But they don't tell you why. And the most logical assumption is: "this type of attribute is not allowed on this type of tag". 
Which is wrong. The warning should be more specific, like... you can't place `aria-multiselectable` on an element without a role of X, Y, Z. 

Anyways, we all learned a bit more about accessibility and a fix, changing the `role` on `<table>` elements to `grid` from `table` will be shipped in next release.

@jackmu95: food for thought (possible enhancement): if user provides a `role` for the element, BV should not override it. The default one should only be applied when no role (even if invalid by today's standard) was provided. I'll look deeper into this.

**Comment by jackmu95** [comment]
> @jackmu95: food for thought (possible enhancement): if user provides a `role` for the element, BV should not override it. The default one should only be applied when no role (even if invalid by today's standard) was provided. I'll look deeper into this.

Good catch! Feel free to PR ist or I will do 😄

**Comment by derick-montague** [comment]
Thank you!
